pkgs/build-support/fetch9front/default.nix

```diff
@@ -0,0 +1,37 @@
+{ fetchgit, fetchzip, lib }:
+
+lib.makeOverridable (
+  { owner
+  , repo
+  , rev
+  , protocol ? "git"
+  , domain ? "git.9front.org"
+  , name ? "source"
+  , leaveDotGit ? false
+  , deepClone ? false
```

Comment:
Don't we need to assert that both are not true according to line 24?

Reply:
Both of these essentially accomplish the same thing, if either of them are requested they result in the same behavior which is cloning the repo over git. Since the upstream server does not support shallow clones, if you request to leaveDotGit we also are required to a deep clone. If that makes sense.